### ISSUE TITLE
Support Function<String,Verifier> for kid mapping

### ISSUE THREAD

**Issue by agentgt** [FusionAuth/fusionauth-jwt#47]
Hi. Great library!

As we were discussing in #46 on mapping kid to key it would be great if you supported a Function<String, Verifier> instead of a Map<String, Verifier>.

```java
 public JWT decode(String encodedJWT, Function<String, Verifier> verifiers)
```

For now we have a hack:

```java
         Function<String, @Nullable PublicKey> keyFunction ; //snip
         var verifiers = new AbstractMap<String, @Nullable Verifier>() {

            @Override
            public @Nullable Verifier get(
                    @Nullable Object key) {
                var k = keyFunction.apply((String) key);
                if (k == null)
                    return null;
                return RSAVerifier.newVerifier(k);
            }

            @Override
            public boolean isEmpty() {
                return false;
            }

            @Override
            public Set<Entry<String, Verifier>> entrySet() {
                throw new UnsupportedOperationException();
            }

        };
```

Obviously this isn't ideal particularly if the original decode method that takes Map calls other things besides `isEmpty`.

Because we have the hack it isn't urgent.

**Comment by robotdan** [comment]
Thanks for the request @agentgt. We'll take a look, you are also welcome to submit a PR. Thanks!

**Comment by robotdan** [comment]
Should be available in 5.2.0, let me know how it works for you.
